src/OnboardingSPA/components/Sidebar/components/Customize/DesignColorsPanel/index.js

```diff
@@ -0,0 +1,247 @@
+import { useState, useEffect } from '@wordpress/element';
+import { useSelect, useDispatch } from '@wordpress/data';
+import { PanelBody, PanelRow, Button } from '@wordpress/components';
+import ColorPickerButton from '../../../../ColorPickerButton';
+import ColorPaletteIcon from './ColorPaletteIcon';
+import CustomColorPalette from './CustomColorPalette';
+import './stylesheet.scss';
+import { store as nfdOnboardingStore } from '../../../../../store';
+// import { useGlobalStylesOutput } from '../../../../../utils/global-styles/use-global-styles-output';
+
+const DesignColorsPanel = ( {
+	baseClassName = 'nfd-onboarding-sidebar-customize--design-colors-panel',
+	heading,
+} ) => {
+	const { storedPreviewSettings, customizeSidebarData } = useSelect(
+		( select ) => {
+			return {
+				storedPreviewSettings:
+					select( nfdOnboardingStore ).getPreviewSettings(),
+				customizeSidebarData:
+					select( nfdOnboardingStore ).getCustomizeSidebarData(),
+			};
+		},
+		[]
+	);
+
+	const { currentData } = useSelect( ( select ) => {
+		return {
+			currentData:
+				select( nfdOnboardingStore ).getCurrentOnboardingData(),
+		};
+	} );
+
+	const design = customizeSidebarData?.design;
+	const colorPalettes = customizeSidebarData?.colorPalettes;
+	const palettePrimaryColors = Object.entries( design?.color_palette ).map(
+		( [ , color ] ) => ( {
+			name: 'Custom',
+			color,
+		} )
+	);
+
+	const defaultColors = {
+		primary: design?.color_palette.primary,
+		secondary: design?.color_palette.secondary,
+		tertiary: design?.color_palette.tertiary,
+	};
+	const palette1 = {
+		primary: colorPalettes[ 1 ]?.primary,
+		secondary: colorPalettes[ 1 ]?.secondary,
+		tertiary: colorPalettes[ 1 ]?.tertiary,
+	};
+	const palette2 = {
+		primary: colorPalettes[ 2 ]?.primary,
+		secondary: colorPalettes[ 2 ]?.secondary,
+		tertiary: colorPalettes[ 2 ]?.tertiary,
+	};
+	const palette3 = {
+		primary: colorPalettes[ 3 ]?.primary,
+		secondary: colorPalettes[ 3 ]?.secondary,
+		tertiary: colorPalettes[ 3 ]?.tertiary,
+	};
```

Comment:
We should move to using a `for` loop and track these palettes in an array instead of separate variables. We may not always have these indexes.

Reply:
Done